src/core/Akka/Serialization/Serializer.cs

```diff
@@ -168,6 +168,15 @@ public sealed override object FromBinary(byte[] bytes, Type type)
         /// <param name="o">The object for which the manifest is needed.</param>
         /// <returns>The manifest needed for the deserialization of the specified <paramref name="o"/>.</returns>
         public abstract string Manifest(object o);
+
+        /// <summary>
+        /// Returns the manifest (type hint) that will be provided in the <see cref="FromBinary(byte[],System.Type)"/> method.
+        /// </summary>
+        /// <param name="type">The <see cref="Type"/> for which the manifest is needed.</param>
+        /// <returns>The manifest needed for the deserialization of the specified <paramref name="type"/>.</returns>
+        public virtual string Manifest(Type type)
```

Comment:
This might not be an ideal solution, would love to hear any input on this one

Reply:
I'd prefer if we didn't do this